planner:
  sot_type: "SoTStatic"
  tasks:
  - name: "EE Pos 1"
    planner_type: "WaypointPlanner"
    ee_pose: [0.0, 1.0, 0.0, 0.0, 0.0, 0.0]  # [x, y, z, roll, pitch, yaw]
    hold_period: 3.0
    tracking_pos_err_tol: 0.02

simulation:
  robot:
    home: [0, -0, 0, 0.25pi, -0.25pi, 0.5pi, -0.25pi, 0.5pi, 0.417pi]  # standard
  static_obstacles:
    enabled: True

controller:
  home: "default"
  self_collision_avoidance_enabled: False
  static_obstacles_collision_avoidance_enabled: True

  collision_constraints_softened:
    self: False
    static_obstacles: False
  collision_constraint_type: "SignedDistanceConstraint"
  type: "MPC"
  acados:
    ocp_solver_options:
      nlp_solver_max_iter: 100
      qp_solver_warm_start: 1
    slack_enabled:
      x: True
      x_e: True
      h_0: True
      h: True
      h_e: False

include:
  - package: "mm_run"
    path: "config/robot/thing.yaml"
  - package: "mm_run"
    path: "config/controller/MPC.yaml"
  - package: "mm_run"
    path: "config/sim/simulation.yaml"
  - package: "mm_run"
    path: "config/scene/simple_obstacles.yaml"

logging:
  log_dir: "3d_collision"
  log_level: 10 # 0 - not set, 10 - debug, 20 - info
